### PR TITLE
travel obf resource manager reader added

### PR DESCRIPTION
**Pull request by angry010101**

### CODE REVIEW

File: OsmAnd/src/net/osmand/plus/resources/ResourceManager.java [hunk 1]
```diff
@@ -1132,12 +1134,29 @@ public synchronized void close(){
 		return readers.toArray(new BinaryMapIndexReader[0]);
 	}
 
+	public BinaryMapIndexReader[] getTravelFiles() {
+		Collection<BinaryMapReaderResource> fileReaders = getFileReaders();
+		List<BinaryMapIndexReader> readers = new ArrayList<>(fileReaders.size());
+		for (BinaryMapReaderResource res : fileReaders) {
+			if (!res.filename.toString().toLowerCase().contains("wiki")) {
```
Comment (MadWasp79) [inline OsmAnd/src/net/osmand/plus/resources/ResourceManager.java:1141]: wiki? I thought it should be "travel"

File: OsmAnd/src/net/osmand/plus/resources/ResourceManager.java [hunk 2]
```diff
@@ -1132,6 +1134,24 @@ public synchronized void close(){
 		return readers.toArray(new BinaryMapIndexReader[0]);
 	}
 
+	public BinaryMapIndexReader[] getTravelFiles() {
+		Collection<BinaryMapReaderResource> fileReaders = getFileReaders();
+		List<BinaryMapIndexReader> readers = new ArrayList<>(fileReaders.size());
+		for (BinaryMapReaderResource res : fileReaders) {
+			if (!res.filename.toString().toLowerCase().contains("travel.obf")) {
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/resources/ResourceManager.java:1141]: IndexConstants for constants

File: OsmAnd/src/net/osmand/plus/resources/ResourceManager.java [hunk 3]
```diff
@@ -1132,6 +1134,24 @@ public synchronized void close(){
...
+				continue;
+			}
+			BinaryMapIndexReader index = res.getReader(BinaryMapReaderResourceType.POI);
+			for (BinaryIndexPart p : index.getIndexes()) {
+				if (p instanceof BinaryMapPoiReaderAdapter.PoiRegion) {
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/resources/ResourceManager.java:1146]: @alex-osm why is it needed

File: OsmAnd/src/net/osmand/plus/wikivoyage/data/TravelObfHelper.java [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+		if (!file.exists()) {
+			GPXUtilities.writeGpxFile(file, gpx);
+		}
+		return file;
+	}
+
+	@Override
+	public String getSelectedTravelBookName() {
+		//TODO REPLACE
+		return "OBF TRAVEL BOOK";
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/wikivoyage/data/TravelObfHelper.java:160]: null

File: OsmAnd/src/net/osmand/plus/wikivoyage/data/TravelDbHelper.java [hunk 5]
```diff
@@ -114,6 +114,18 @@ public TravelDbHelper(OsmandApplication application) {
 		localDataHelper = new TravelLocalDataHelper(application);
 	}
 
+	public static boolean checkIfDbFileExists(OsmandApplication app) {
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/wikivoyage/data/TravelDbHelper.java:117]: Duplicate to checkIfObfFileExists

File: OsmAnd/src/net/osmand/plus/wikivoyage/data/TravelObfHelper.java [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+
+	private static final Log LOG = PlatformUtil.getLog(TravelObfHelper.class);
+
+	private static final int POPULAR_LIMIT = 25;
+
+	private final OsmandApplication application;
+
+	private TravelLocalDataHelper localDataHelper;
+
+	private List<BinaryMapIndexReader> files;
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/wikivoyage/data/TravelObfHelper.java:38]: ResourceManager

File: OsmAnd/src/net/osmand/plus/wikivoyage/data/TravelObfHelper.java [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+	private List<File> existingTravelBooks = new ArrayList<>();
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/wikivoyage/data/TravelObfHelper.java:39]: Why it is needed?

File: OsmAnd/src/net/osmand/plus/wikivoyage/data/TravelObfHelper.java [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+				if (f.getName().contains(IndexConstants.BINARY_TRAVEL_GUIDE_MAP_INDEX_EXT)) {
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+
+	// might use in future
+	protected static class PopularArticle {
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/wikivoyage/data/TravelObfHelper.java:176]: Delete not used class

**Pull request merged**